src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -17,10 +17,227 @@ namespace System
+
+                    // The Slice is necessary since startIndex cannot equal source.Length here.
+                    source.Slice(0, startIndex).CopyTo(destination);
+                }
+                else
+                {   // no overlap or destination is before source
+                    // The Slice is necessary since startIndex cannot equal source.Length here.
+                    source.Slice(0, startIndex).CopyTo(destination);
+
+                    if (startIndex + count != source.Length)
+                        source.Slice(startIndex + count).CopyTo(destination.Slice(startIndex));
+                }
+            }
+            else
+            {   // Remove everything
+                if (source.Length < destination.Length)
+                    destination.Slice(0, source.Length).Clear();
+                else
+                    destination.Clear();
+            }
+        }
+
+        /// <summary>
+        /// Moves the source characters to the destination and right-aligns them by padding with
+        /// spaces on the left, for a specified total length.
+        /// </summary>
+        /// <param name="source">The source span that contains the characters to be padded.</param>
+        /// <param name="totalWidth">The number of characters in the resulting span, 
+        /// equal to the number of original characters plus any additional padding characters.</param>
+        /// <param name="destination">The span that contains the resulting right-aligned and padded characters.</param>
+        /// <remarks>
+        /// If the totalWidth is less than or equal to the length of the source,
+        /// the entire source is copied into the destination without padding.
+        /// </remarks>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when the destination span is too small to contain the source span with the padding
+        /// (i.e. destination.Length &lt; totalWidth or  destination.Length &lt; source.Length).
+        /// </exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when <paramref name="totalWidth"/> is less than zero.
+        /// </exception>
+        public static void PadLeft(this ReadOnlySpan<char> source, int totalWidth, Span<char> destination)
+        {
+            if (totalWidth < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+            if (destination.Length < totalWidth)
+                ThrowHelper.ThrowArgumentException_DestinationTooShort();
+
+            int count = totalWidth - source.Length;
+            if (count <= 0)
+            {
+                // This will throw if destination.Length < source.Length
+                source.CopyTo(destination);
+            }
+            else
+            {
+                source.CopyTo(destination.Slice(count));
+                for (int i = 0; i < count; i++)
```

Comment:
Would it be faster to use Fill?

Reply:
It was slower to do so, especially for small amounts, similar to https://github.com/dotnet/corefx/pull/26289#discussion_r162713305.

We also avoid an unnecessary Slice.

Reply:
What does that say about our Fill API is we can't use it? Do we successfully use it anywhere you've done per measurements? Maybe Fill should just be fixed.

Reply:
https://github.com/dotnet/corefx/issues/26604